### DISCUSSION TITLE
Question regarding Year on Year calculation

### DISCUSSION THREAD

**Discussion by frankspostfach** [sql-bi/Bravo#483] [Q&A] [answered]
Hi,

I am a new user of "Bravo". First: Many thanks for this tool! It makes first steps with Power BI and DAX so much easier.

My question is regarding the **Time Intelligence function** in Bravo. I noticed that the **"YOY Measures"** will not show values in my reports, if Previous Year is empty. From my understanding, this is due to DAX and the use of "NOT ISBLANK". 

But it will lead to "wrong" results on detail level, while the Totals are correct. 
I am just curious if this is on purpose? If I am changing DAX of the YOY Measure, every time I am updating via Bravo it will be overwritten?

![result](https://user-images.githubusercontent.com/113419906/189836517-86737cee-e5aa-48b1-a528-7dd5c20875e5.png)

**Comment by marcosqlbi** [comment, marked as answer]
The current template we use is described here (there are different patterns depending on the Date table you choose): https://www.daxpatterns.com/time-patterns/

The current logic is simple to implement but does not work well when you have the situation described in your screenshot.
 I think that the PY and the YOY% should still be blank, but at this point seeing the YOY alone with a copy of the measure would be confusing - or do you think it would be better? You can customize the template by changing the logic implemented in YOY Measures (see https://docs.sqlbi.com/bravo/features/manage-dates/customize-date-template - we are also working on videos we'll integrate in the documentation before releasing v1) otherwise your change to the measure will be overwritten if you apply the template again with Bravo.

**Comment by frankspostfach** [comment]
Hi & thanks for your answer!

To illustrate my point see picture attached: The problem with this kind of YOY calculation, that I don't see growth in the year on year column, if the previous year is blank (and if actual year is blank and PY has values) Do you see my point? And as you said: If I customize DAX formula, every time I apply for update of the model, my changes are overwritten. I am surprised that nobody brought up this before.

Or am I wrong? 

![image](https://user-images.githubusercontent.com/113419906/191675383-6dfc6255-ef0e-4693-9c92-a16e8686bae1.png)

**Comment by marcosqlbi** [comment]
I completely understand your point. However, in a generic template, the alternative is to always show the YOY value, which would result in a confusing report in other scenarios.
As I said in my previous reply, you can now customize the template so that it will always apply your logic (also in other models) instead of the built-in one.
For the generic template, I will make some tests in changing the YOY without changing PY and YOY%, but don't rely on it because I might decide to keep the existing behavior. Go ahead and customize the template according to your requirements.
